### PR TITLE
release: v4.5.3

### PR DESCRIPTION
**Pull request by arya2**
Hotfix release **Zebra 4.5.3**.

Adds a soft fork that temporarily disables Orchard actions in transactions to mitigate a security issue ([GHSA-jfw5-j458-pfv6](https://github.com/ZcashFoundation/zebra/security/advisories/GHSA-jfw5-j458-pfv6)), activating at Mainnet height **3,363,426**.

- Version bumps: `zebra-chain` 8.0.1, `zebra-consensus`/`zebra-network`/`zebra-state` 7.0.1, `zebrad` 4.5.3.
- CHANGELOG + README updated.
- Checkpoints/end-of-support intentionally not refreshed: v4.5.1 shipped 3 days ago, and the activation height (3,363,426) is above the highest checkpoint (3,358,006), so the rule is enforced via semantic verification.

Release is performed from this branch per the hotfix checklist.

### CODE REVIEW

**Review by oxarbitrage — COMMENTED** [review]
Reviewed commit-by-commit. Core Orchard enforcement logic is sound — the `orchard_shielded_data().is_some()` check correctly rejects V5/V6 transactions with Orchard data, tests are thorough (boundary, block, mempool paths), version bumps are correct, CHANGELOG is accurate.

**Critical issue in commit `4434a6c0` — causing CI failure:**

In `zebra-state/src/service/chain_tip.rs`, this line was changed from `block.height` to `block.height.next().unwrap()`:

```rust
|| NetworkUpgrade::is_activation_height(&self.network, block.height.next().unwrap())
```

This shifts ALL network upgrade mempool resets to fire one block early (at height N-1 instead of N), not just the Orchard soft fork. The CI proptest `best_tip_is_latest_non_finalized_then_latest_finalized` fails because at Height(3146399), `next()` returns 3146400 (Nu6 activation), triggering a spurious `Reset` instead of `Grow`.

```
left:  Some(Grow  { height: Height(3146400) ... })
right: Some(Reset { height: Height(3146400) ... })
```

Fix: revert to `block.height` for the `is_activation_height` call. The new soft fork check on the adjacent line should also use `block.height` directly.

Minor note: commits 3 (`4434a6c0`) and 5 (`efca5e8f`) don't compile independently — commit 3 references methods that only exist in commit 5. Worth squashing for bisectability.

File: zebra-state/src/service/chain_tip.rs [hunk 1]
```diff
@@ -595,8 +595,18 @@ impl ChainTipChange {
...
+        // that are no longer valid under the new rule. So we also reset there.
         if Some(block.previous_block_hash) != self.last_change_hash
-            || NetworkUpgrade::is_activation_height(&self.network, block.height)
+            || NetworkUpgrade::is_activation_height(&self.network, block.height.next().unwrap())
+            || self
+                .network
+                .is_temporary_orchard_disabling_soft_fork_activation_height(
+                    block.height.next().unwrap(),
+                )
         {
```
Comment (gustavovalverde) [inline zebra-state/src/service/chain_tip.rs:610]: Blocker, as it breaks CI: This also rewrites the existing NetworkUpgrade reset (`block.height` -> `block.height.next().unwrap()`). `is_activation_height` is exact-match, so the NU reset now fires at activation-1, not at the activation block: a silent ZIP-200 timing change for every upgrade.

It breaks `best_tip_is_latest_non_finalized_then_latest_finalized` (oracle at `tests/prop.rs:135` still uses `chain_tip.height`).

`.next()` is right for the soft-fork term only (mempool verifies at tip+1, so Orchard txs must drop at activation-1). Revert the NU term, compute next once:

```suggestion
        let next_height = block
            .height
            .next()
            .expect("chain tip height is far below Height::MAX");
        if Some(block.previous_block_hash) != self.last_change_hash
            || NetworkUpgrade::is_activation_height(&self.network, block.height)
            || self
                .network
                .is_temporary_orchard_disabling_soft_fork_activation_height(next_height)
        {
```

Companion `tests/prop.rs` (not in the diff, so not inline-suggestable):

```rust
} else if Some(chain_tip.previous_block_hash) != old_last_change_hash
    || NetworkUpgrade::is_activation_height(&network, chain_tip.height)
    || chain_tip.height.next().is_ok_and(|next| {
        network.is_temporary_orchard_disabling_soft_fork_activation_height(next)
    })
{
```

With both together, the test passes (verified locally). Also, it seems valid to add a vectors test asserting `Reset` at the soft-fork boundary and to fix the comment at 599-602 (soft-fork resets one block before activation; NU at it).

File: zebra-consensus/src/transaction.rs [hunk 2]
```diff
@@ -406,6 +406,18 @@ where
...
+            // Soft fork: temporarily require transactions to not contain Orchard actions.
+            //
+            // This soft fork was added while NU 6.1 was the active epoch on the Zcash
+            // chain, but we apply it uniformly even if NU 6.1 is not active in case it is
+            // ported to other chains with a different sequence of NUs.
+            //
+            // This will be treated as "Rules that apply generally before the next NU"
+            // when we add the NU that re-enables Orchard actions.
+            if network.temporary_orchard_disabling_soft_fork_active(req.height()) && tx.orchard_shielded_data().is_some() {
+                return Err(TransactionError::Other("transaction has Orchard actions (temporarily disabled)".into()));
```
Comment (gustavovalverde) [inline zebra-consensus/src/transaction.rs:418]: Nit: `TransactionError::Other` scores 0 in `mempool_misbehavior_score()` (`_other => 0`), so peers spamming now-invalid Orchard txs post-activation get no penalty (vs 100 for `NotEnoughFlags`/`WrongConsensusBranchId`). Reusing `Other` is fine for a hotfix; we should consider whether a follow-up could add a scored variant or something similar.

File: zebra-network/src/config.rs [hunk 3]
```diff
@@ -607,6 +607,11 @@ struct DTestnetParameters {
     /// If `true`, automatically repeats configured funding stream addresses to fill
     /// all required periods.
     extend_funding_stream_addresses_as_required: Option<bool>,
+    /// Height at which the soft fork that temporarily disables Orchard actions activates.
+    ///
+    /// If unset, the default activation height for the network is used; the soft fork
+    /// cannot be disabled via configuration.
```
Comment (gustavovalverde) [inline zebra-network/src/config.rs:613]: Nit: `cannot be disabled via configuration` fits Mainnet's fixed height, but not here, where this field defaults to `None` (off) and is the per-Testnet enable switch, I think it's worth distinguishing the two.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,21 @@ All notable changes to Zebra are documented in this file.
...
+This hotfix release adds a soft fork that temporarily disables Orchard actions in
+transactions, to mitigate a security issue. We recommend node operators update to
+4.5.3 as soon as possible.
+
+### Security
+
+- Add a soft fork that temporarily rejects transactions containing Orchard actions.
+  It activates at a fixed height on Mainnet and at a configurable height on Testnet
+  (`temporary_orchard_disabling_soft_fork_height` in the network config), and the
+  mempool is revalidated at the activation height to drop any Orchard transactions
```
Comment (gustavovalverde) [inline CHANGELOG.md:19]: Nit: With the reset firing at activation-1 (see `chain_tip.rs`), the mempool is revalidated when the block before activation lands, not `at the activation height`. Suggestion: `just before the activation height`.

**Review by gustavovalverde — CHANGES_REQUESTED** [review]
One blocker: the `chain_tip.rs` change also rewrites the existing NetworkUpgrade reset, breaking the `chain_tip` prop test.

**Comment by conradoplg** [comment]
Admin-merging with a merge commit per the hotfix checklist. Issues identified will be addressed in a later PR.

**Comment by oxarbitrage** [comment]
Revisiting my earlier review comment on `block.height.next().unwrap()` — after reviewing #10676 (`fix(state,zebrad): fix tests`) and conrado's explanation, the `next()` semantics are correct. The mempool verifies transactions against the next block height, so the reset needs to fire at `ACTIVATION_HEIGHT - 1` (when the next block to be mined would be the activation block). My earlier call to revert to `block.height` was wrong. #10676 fixes the tests and clarifies the intent in the comments.
